en/cpp/guide/connections.md

```diff
@@ -1,39 +1,100 @@
 Once monitoring a port, MAVSDK will automatically detect connected vehicles, add them to its collection, and notify registered users of connection and disconnection events.
 
 ## Monitoring a Port
 
-Specify the port(s) to watch using one of the (synchronous) connection methods: [add_any_connection()](../api_reference/classmavsdk_1_1_mavsdk.md#classmavsdk_1_1_mavsdk_1a405041a5043c610c86540de090626d97), [add_udp_connection()](../api_reference/classmavsdk_1_1_mavsdk.md#classmavsdk_1_1_mavsdk_1aa43dfb00d5118d26ae5aabd0f9ba56b2), [add_tcp_connection()](../api_reference/classmavsdk_1_1_mavsdk.md#classmavsdk_1_1_mavsdk_1a91c7a70c6e8ffa43844f2ce04f2696f0) or [add_serial_connection()](../api_reference/classmavsdk_1_1_mavsdk.md#classmavsdk_1_1_mavsdk_1a669ddeec7af571fdbde9f31e343d50ac).
-All the methods are used similarly, and return immediately with a [ConnectionResult](../api_reference/namespacemavsdk.md#namespacemavsdk_1a0bad93f6d037051ac3906a0bcc09f992) indicating whether they succeeded.
+Specify the port(s) to watch using one of the (synchronous) connection method: [add_any_connection()](../api_reference/classmavsdk_1_1_mavsdk.md#classmavsdk_1_1_mavsdk_1a405041a5043c610c86540de090626d97). The method returns immediately with a [ConnectionResult](../api_reference/namespacemavsdk.md#namespacemavsdk_1a0bad93f6d037051ac3906a0bcc09f992) indicating whether it succeeded.
 
-The [add_any_connection()](../api_reference/classmavsdk_1_1_mavsdk.md#classmavsdk_1_1_mavsdk_1a405041a5043c610c86540de090626d97) method can be used to set up monitoring for any of the supported port types (while the other methods set up specific connection types).
 The connection details are specified using the string formats shown below:
 
 Connection | URL Format
 --- | ---
-UDP | `udp://[Bind_host][:Bind_port]`
-TCP | `tcp://[Server_host][:Server_port]`
-Serial | `serial://[Dev_Node][:Baudrate]`
+UDP | `udp://[host][:port]`
+TCP | `tcp://[host][:port]`
+Serial | `serial://[path][:baudrate]`
 
-The code snippet below shows how to set up monitoring with `add_any_connection()`:
+### Connecting over serial
+
+To add a serial connection (e.g. over USB, FTDI, or an SiK radio), you specify the serial port and the baudrate like this:
+
+**On Linux:**
+
+```cpp
+Mavsdk mavsdk;
+ConnectionResult connection_result = mavsdk.add_any_connection("serial:///dev/serial/by-id/usb-FTDI_FT232R_USB_UART_XXXXXXXX-if00-port0:57600");
+if (connection_result != ConnectionResult::Success) {
+    std::cout << "Adding connection failed: " << connection_result << '\n';
+    return;
+}
+```
+
+**On Windows:**
+
+```cpp
+Mavsdk mavsdk;
+ConnectionResult connection_result = mavsdk.add_any_connection("serial://COM3:57600");
+if (connection_result != ConnectionResult::Success) {
+    std::cout << "Adding connection failed: " << connection_result << '\n';
+    return;
+}
+```
+
+
+### Connecting over UDP
+
+When connecting over UDP, there are two setups to distinguish: server and client mode.

```

Comment:
So essentially the normal mode we use for SDK to act like a server (GCS) and listen. We can ONLY do this for the local host right? You can't specify that you want to listen on some other interface?

Reply:
Correct.